openedx/core/djangoapps/schedules/signals.py

```diff
@@ -27,59 +27,31 @@
 
 
 @receiver(post_save, sender=CourseEnrollment, dispatch_uid='create_schedule_for_enrollment')
-def create_schedule(sender, **kwargs):
-    if not kwargs['created']:
-        # only create schedules when enrollment records are created
-        return
-
-    current_site = get_current_site()
-    if current_site is None:
-        log.debug('Schedules: No current site')
-        return
-
-    enrollment = kwargs['instance']
-    schedule_config = ScheduleConfig.current(current_site)
-    if (
-        not schedule_config.create_schedules
-        and not CREATE_SCHEDULE_WAFFLE_FLAG.is_enabled(enrollment.course_id)
-    ):
-        log.debug('Schedules: Creation not enabled for this course or for this site')
-        return
-
-    if not enrollment.course_overview.self_paced:
-        log.debug('Schedules: Creation only enabled for self-paced courses')
-        return
-
-    # This represents the first date at which the learner can access the content. This will be the latter of
-    # either the enrollment date or the course's start date.
-    content_availability_date = max(enrollment.created, enrollment.course_overview.start)
-
-    upgrade_deadline = _calculate_upgrade_deadline(enrollment.course_id, content_availability_date)
-
-    if course_has_highlights(enrollment.course_id):
-        experience_type = ScheduleExperience.EXPERIENCES.course_updates
-    else:
-        experience_type = ScheduleExperience.EXPERIENCES.default
-
-    if _should_randomly_suppress_schedule_creation(
-        schedule_config,
-        enrollment,
-        upgrade_deadline,
-        experience_type,
-        content_availability_date,
-    ):
-        return
-
-    schedule = Schedule.objects.create(
-        enrollment=enrollment,
-        start=content_availability_date,
-        upgrade_deadline=upgrade_deadline
-    )
-
-    ScheduleExperience(schedule=schedule, experience_type=experience_type).save()
-
-    log.debug('Schedules: created a new schedule starting at %s with an upgrade deadline of %s and experience type: %s',
-              content_availability_date, upgrade_deadline, ScheduleExperience.EXPERIENCES[experience_type])
+def create_schedule(_, **kwargs):
+    """
+    When a CourseEnrollment is created, create a Schedule if configured
+    """
+    try:
+        enrollment = kwargs['instance']
```

Comment:
Nit: did you want this to be `kwargs.get`?  Since you later check for `if enrollment` in the `except` clause?